utils/browser/src/lib.rs

```diff
@@ -0,0 +1,177 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+use futures01::sync::mpsc as mpsc01;
+use log::{debug, info};
+use std::{path::Path, sync::Arc};
+use service::{
+    AbstractService, RpcSession, Roles, Configuration, config::DatabaseConfig,
+    ChainSpec, RuntimeGenesis
+};
+use wasm_bindgen::prelude::*;
+use futures::{
+    TryFutureExt as _, FutureExt as _, Stream as _, Future as _, TryStreamExt as _,
+    channel::{oneshot, mpsc}, future::{poll_fn, ok}, compat::*,
+};
+use std::task::Poll;
+use std::pin::Pin;
+use chain_spec::Extension;
+
+pub use libp2p::wasm_ext::{ExtTransport, ffi::Transport};
+
+/// Set up a panic hook and console logging.
+pub fn set_hooks(log_level: log::Level) {
+    console_error_panic_hook::set_once();
+    let _ = console_log::init_with_level(log_level);
```

Comment:
It might be a good idea to re-export these functions instead.

Reply:
Yeah, not sure this makes a lot of sense to me. All this does is encapsulate two functions into one and swallow an error. First wanted to suggest you return the error ... but then the difference is almost non existent so I wonder if this function is that useful...